File: study/22.09.20/EJH/k진수에서 소수 개수 구하기.py
import math


def change_num(n, k):
    num = 0
    s = 1
    P = []
    while n > 0:
        num += (n % k) * s
        if n % k == 0:
            if num != 0:
                P.append(num)
            s = 1
            num = 0
        else:
            s *= 10
        n //= k

    if num != 0:
        P.append(num)

    return P


def check_prime(n):
    if n == 1:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True


def solution(n, k):
    answer = 0

    P = change_num(n, k)

    for p in P:
        if check_prime(p):
            answer += 1
    return answer


print(solution(437674,	3))
print(solution(110011,	10))

File: study/22.09.20/HHS/K진수에서소수개수구하기.java
import java.util.*;
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        
        //k진수로 바꾸기
        //String으로 받아서 0을 기준으로 자르기
        //잘린 문자들을 숫자로 받아서 소수인지 판별 후 answer 올리기
        String KString = "";
        while(n > 0){
            KString = (n % k)+KString;
            n /= k;
        }
        StringTokenizer token = new StringTokenizer(KString, "0");
        while(token.hasMoreTokens()){
            long num = Long.parseLong(token.nextToken());
            if(isPrime(num))
                answer++;
        }
        
        
        return answer;
    }
    private static boolean isPrime(long num){
        if(num == 1) return false;
        if(num == 2) return true;

        for(int i = 3; i <= Math.sqrt(num); i++){
            if(num % i == 0)
                return false;
        }
        return true;
    }
}